src/commands/basemaps-github/make.cog.github.ts

```diff
@@ -160,6 +169,96 @@ export class MakeCogGithub {
     return tileSet;
   }
 
+  /**
+   * Given a old and new lds layer stac item and log the changes for pull request body
+   */
+  getVectorChanges(newLayer: StacLink | undefined, existingLayer: StacLink | undefined): string | null {
+    // Update Layer
+    if (newLayer != null && existingLayer != null) {
+      const featureChange = Number(newLayer['lds:feature_count']) - Number(existingLayer['lds:feature_count']);
+
+      if (newLayer['lds:version'] === existingLayer['lds:version'] && featureChange !== 0) {
+        // Alert if feature changed with no version bump.
+        return `🟥🟥🟥🟥 Feature Change Detected ${newLayer['lds:name']} - version: ${newLayer['lds:version']} features: ${newLayer['lds:feature_count']} (+${featureChange}) 🟥🟥🟥🟥`;
+      }
+
+      if (featureChange >= 0) {
+        // Add Features
+        return `🟦 ${newLayer['lds:name']} - version: ${newLayer['lds:version']} (from: ${existingLayer['lds:version']}) features: ${newLayer['lds:feature_count']} (+${featureChange})`;
+      } else {
+        // Remove Features
+        return `🟧 ${newLayer['lds:name']} - version: ${newLayer['lds:version']} (from: ${existingLayer['lds:version']}) features: ${newLayer['lds:feature_count']} (-${featureChange})`;
+      }
+    }
+
+    // Add new Layer
+    if (newLayer != null && existingLayer == null) {
+      return `🟩 ${newLayer['lds:name']} - version: ${newLayer['lds:version']} features: ${newLayer['lds:feature_count']}`;
+    }
+
+    // Remove Layer
+    if (newLayer == null && existingLayer != null) {
+      return `🟥 ${existingLayer['lds:name']} features: -${existingLayer['lds:feature_count']}`;
+    }
+
+    // No changes detected return null
+    return null;
+  }
+
+  /**
+   * Prepare and create pull request for the aerial tileset config
+   */
+  async diffVectorUpdate(layer: ConfigLayer, existingTileSet?: ConfigTileSetVector): Promise<string | undefined> {
+    const changes: (string | null)[] = [];
+    // Vector layer only support for 3857
+    if (layer[3857] == null) return;
+    const newCollectionPath = new URL('collection.json', layer[3857]).href;
+    const newCollection = await fsa.readJson<StacCollection>(newCollectionPath);
+    if (newCollection == null) throw new Error(`Failed to get target collection json from ${newCollectionPath}.`);
+    const ldsLayers = newCollection.links.filter((f) => f.rel === 'lds:layer') as StacLinkLds[];
+
+    // Log all the new inserts for new tileset
+    if (existingTileSet == null) {
+      changes.push(`New TileSet ts_${layer.name} with layer ${layer.name}.\n`);
+      for (const l of ldsLayers) {
+        changes.push(this.getVectorChanges(l, undefined));
+      }
+      return changes.filter((f) => f != null).join('\n');
+    }
+
+    // Compare the different of existing tileset, we usually only have one layers in the vector tiles, so the loop won't fetch very much
+    for (const l of existingTileSet.layers) {
+      if (l[3857] == null) continue;
```

Comment:
This magic value should probably be a constant.

Reply:
if you hover over `l` this shows the type as a `EpsgCode`.

The EpsgCode `3857` in our code base is generally just `3857`, the naming of it in some places is shown as `Epsg.WebMercator` or `Epsg.Google` but these both make it much harder to understand the actual code and then you need more knowledge of our system to know that `Epsg.Google` is referring back to 3857.